src/Workspaces/Core/Portable/Remote/IRemoteServiceCallbackDispatcherProvider.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Microsoft.CodeAnalysis.Remote
+{
+    internal interface IRemoteServiceCallbackDispatcherProvider
+    {
+        IRemoteServiceCallbackDispatcher GetDispatcher(Type serviceType);
```

Comment:
😕 This seems like something the partner would manage as a completely independent brokered service. The only part they need from us is a brokered service that gives access to a specific solution.

Reply:
We could remove these but then everyone would need to reimplement pretty much the same logic we have in Roslyn. I opted to make that infrastructure available, so it's easy to use ISB for partners.

Reply:
We also don't want to expose the different descriptors for 32/64/64S.